trunner/harness/unity.py

```diff
@@ -35,7 +35,7 @@ def unity_harness(dut: Dut, ctx: TestContext) -> Optional[TestResult]:
     result_re = r"TEST\((?P<group>\w+), (?P<name>\w+)\) (?P<status>PASS|IGNORE)"
     # Fail need to have its own regex due to greedy matching
     result_final_re = r"TEST\((?P<group>\w+), (?P<name>\w+)\) (?P<status>FAIL) at (?P<path>.*?):(?P<line>\d+)\r"
-    final_re = r"(?P<total>\d+) Tests (?P<fail>\d+) Failures (?P<ignore>\d+) Ignored"
+    final_re = r"(?P<total>\d+) Tests (?P<fail>\d+) Failures (?P<ignore>\d+) Ignored [\r\n]+(?P<result>OK|FAIL)"
```

Comment:
There could be double \r, that's why we commonly use \r+. I think it's rather not probable to have 2 newlines there, so []+ is not necessary in my opinion.

```suggestion
    final_re = r"(?P<total>\d+) Tests (?P<fail>\d+) Failures (?P<ignore>\d+) Ignored \r+\n(?P<result>OK|FAIL)"
```